src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -158,14 +158,29 @@ def invoke_package_manager(self, command):
             else:  # verbose diagnostic log
                 self.log_success_on_invoke(code, out)
 
-            if code == self.zypper_exitcode_zypper_updated:
-                self.composite_logger.log_debug(" - Package manager update detected. Patch installation run will be repeated.")
-                self.set_package_manager_setting(Constants.PACKAGE_MGR_SETTING_REPEAT_PATCH_OPERATION, True)
-            elif code == self.zypper_exitcode_reboot_required:
-                self.composite_logger.log_warning("Machine requires reboot after patch installation. Setting force_reboot flag to True.")
-                self.force_reboot = True
+            if code == self.zypper_exitcode_zypper_updated or code == self.zypper_exitcode_reboot_required:
+                self.__handle_repeat_or_reboot_exit_codes(command, code)
+                
             return out
 
+    def __handle_repeat_or_reboot_exit_codes(self, command, code):
```

Comment:
__handle_zypper_updated_or_reboot_exit_codes

- since repeat is the outcome, and zypper_updated is the code.